components/script/dom/htmlsourceelement.rs

```diff
@@ -46,6 +49,21 @@ impl VirtualMethods for HTMLSourceElement {
         Some(self.upcast::<HTMLElement>() as &VirtualMethods)
     }
 
+    fn attribute_mutated(&self, attr: &Attr, mutation: AttributeMutation) {
+        self.super_type().unwrap().attribute_mutated(attr, mutation);
+        match attr.local_name() {
+            &local_name!("srcset") | &local_name!("sizes")  |
+            &local_name!("media") | &local_name!("type") => {
+                if let Some(next_sibling) = self.upcast::<Node>().GetNextSibling() {
```

Comment:
We will need to iterate over `self.upcast::<Node>().following_siblings()` instead of just getting the immediately following sibling.